src/core/Table/utils.ts

```diff
@@ -29,3 +30,91 @@ export const getCellStyle = <T extends Record<string, unknown>>(
   }
   return style;
 };
+
+const isLastLeftStickyColumn = <T extends Record<string, unknown>>(
+  id: IdType<T>,
+  columns: ColumnInstance<T>[],
+) => {
+  const index = columns.findIndex((x) => x.id === id);
+  const nextColumn = columns[index + 1];
+  return !nextColumn.sticky;
+};
+
+const getLeftMargin = <T extends Record<string, unknown>>(
+  id: IdType<T>,
+  columns: ColumnInstance<T>[],
+) => {
+  let leftMargin = 1;
+  const index = columns.findIndex((x) => x.id === id);
+  for (let i = 0; i < index; i++) {
+    const column = columns[i];
+    if (column.width) {
+      leftMargin += +column.width;
+    }
+  }
+  return leftMargin;
+};
+
+const getRightMargin = <T extends Record<string, unknown>>(
+  id: IdType<T>,
+  columns: ColumnInstance<T>[],
+) => {
+  let rightMargin = 1;
+  const index = columns.findIndex((x) => x.id === id);
+  for (let i = index + 1; i < columns.length; i++) {
+    const column = columns[i];
+    if (column.width) {
+      rightMargin += +column.width;
+    }
+  }
+  return rightMargin;
+};
+
+const isLastRightStickyColumn = <T extends Record<string, unknown>>(
+  id: IdType<T>,
+  columns: ColumnInstance<T>[],
+) => {
+  return columns[columns.length - 1].id === id;
+};
+
+export const getStickyCellStyle = <T extends Record<string, unknown>>(
+  tableInstance: TableInstance<T>,
```

Comment:
And shadow of left and right sticky columns seem to differ

![image](https://user-images.githubusercontent.com/81580355/161236772-7b836c64-b43f-4c9b-8a47-238423ec0ea5.png)

But firstly we will need to review styling in general, how all should look knowing current table structure